src/app/resources/iis_scripts/admin_api_util.ps1

```diff
@@ -66,27 +88,76 @@ function EnsureMember($group, $userOrGroup) {
 ## TODO: Consider creating a lock around this script because it is not concurrency safe
 
 if ($install) {
-    InstallAPI
+    InstallAPI $serviceName
+} else {
+    try {
+        $pingEndpoint = "https://localhost:$adminAPIPort"
+        Write-Verbose "Pinging Admin API at $pingEndpoint"
+        Invoke-WebRequest -UseDefaultCredentials -UseBasicParsing $pingEndpoint | Out-Null
+    } catch {
+        if ($_.Exception.Status -eq [System.Net.WebExceptionStatus]::ConnectFailure) {
+            $service = Get-Service -Name $serviceName -ErrorAction SilentlyContinue
+            if (!$service) {
+                throw "IIS Administration API is not installed"
+            }
+            if ($service.Status -eq [System.ServiceProcess.ServiceControllerStatus]::Running) {
+                throw "IIS Administration API is not found, port pinged: $adminAPIPort"
+            }
+            throw "Unexpected service status for IIS Administration API: ""$($service.Status)"""
+        }
+        throw $_
+    }
 }
 
 $service = Get-WmiObject win32_service | Where-Object {$_.name -eq $serviceName}
-if ($service.StartInfo.EnvironmentVariables -and $service.StartInfo.EnvironmentVariables["USE_CURRENT_DIRECTORY_AS_ROOT"] -and $service.StartInfo.WorkingDirectory) {
-    $workingDirectory = $service.StartInfo.WorkingDirectory
+if ($service) {
+    if ($service.StartInfo.EnvironmentVariables -and $service.StartInfo.EnvironmentVariables["USE_CURRENT_DIRECTORY_AS_ROOT"] -and $service.StartInfo.WorkingDirectory) {
+        $workingDirectory = $service.StartInfo.WorkingDirectory
+    } else {
+        $proc = Get-Process -id $service.ProcessId
+        $workingDirectory = GetIISAdminHome $proc
+    }
 } else {
-    $proc = Get-Process -id $service.ProcessId
-    $workingDirectory = Split-Path ($proc.Modules | Where-Object { $_.ModuleName -eq "Microsoft.IIS.Administration.dll" }).FileName
+    ## dev-mode support, no restart can be perfomed
+    Write-Verbose "Dev mode, scanning processes for IIS Admin API"
+    $devMode = $true
+    $workingDirectory = GetIISAdminHome (Get-Process -ProcessName dotnet)
 }
 
 $configLocation = [System.IO.Path]::Combine($workingDirectory, "config", "appsettings.json")
+if ($devMode -and !(Test-Path $configLocation)) {
+    $configLocation = [System.IO.Path]::Combine($workingDirectory, "..", "..", "..", "config", "appsettings.json")
+}
+Write-Verbose "Config Location $configLocation"

```

Comment:
please add an error handling to check the appsettings.json file exists or not at this point.

Reply:
If the file is not here, all I can do is throw error.
The `Get-Content` line would have thrown error though. Not much difference in user experience.